core/domain/exp_jobs_one_off.py

```diff
@@ -57,6 +60,32 @@ def map(exploration_model):
     def reduce(exp_id, list_of_exps):
         pass
 
+class ExpSummariesContributorsOneOffJob(jobs.BaseMapReduceJobManager):
+    """ One-off job that finds the user id's of the contributors
+    (defined as any human who has made a positive (i.e. not just
+    a reversion) for each exploration.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationSnapshotMetadataModel]
+
+    @staticmethod
+    def map(item):
+        _VERSION_DELIMITER = '-'
```

Comment:
I'm not sure I understand the question. Couldn't you just add something like this to VersionedModel (warning -- I haven't tested it)?

    def get_unversioned_instance_id(self):
        return self.id[:self.id.rfind(_VERSION_DELIMITER)]